File: frontend/src/app/shared/interfaces/MovieListResponse.ts
export interface MovieListResponse {
    currentPage: number
    pageSize: number
    totalResults: number
    currentPageSize:number
    movies: Movie[]
  }
  
  export interface Movie {
    movieId: number
    title: string
    releaseYear: string
    description: string
    thumbnail: string
    duration: number
    language: string
    price:number
    matchedWith:string
}


  

File: backend/MovieMS/src/main/java/com/prime/movie/utility/MovieListResponse.java
package com.prime.movie.utility;

import com.prime.movie.dto.MovieSummaryDto;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Builder
@NoArgsConstructor
@AllArgsConstructor
@Data
public class MovieListResponse {

    Integer currentPage;

    Integer pageSize;
    Long totalResults;
    Integer currentPageSize;

    List<MovieSummaryDto> movies;

}
